### ISSUE TITLE
Improve audio support

### ISSUE THREAD

**Issue by Rinnegatamante** [Rinnegatamante/easyrpg-player-3ds#7]
This is a complete TODO list regarding audio support:

BGM (Musics):
- [ ]  OGG support **(partially implemented)**
- [ ]  MP3 support
- [ ]  MIDI support
- [x]  BGM_Pitch implementation
- [ ]  BGM_GetTicks implementation

Generic:
- [x] Write a proper audio wrapper which uses dsp::DSP service if csnd:SND is unavailable.

**Comment by Ghabry** [comment]
Just to keep the people only following github and not the forum informed:
MP3 and MIDI support will be provided by the EasyRPG devs because the way how MP3/MIDI are currently supported is bad.

And BGM_GetTicks is a MIDI-only feature btw.

**Comment by Rinnegatamante** [comment]
BGM_Pitch produces a lot of bugs atm:

On csnd:SND:

1) When called it seems to not be able to resume correctly sometimes, in the specific it seems to follow this pattern:

1st call: BGM playstate is paused but not in logic streaming (so it will continue the streaming but you'll not listen anything like if volume is 0) at the end of BGM_Pitch.
2nd call: BGM is fine.
3rd call: Same as 1st call.
...

2) Looks like it produces random system crashes if used with pretty high samplerates (?)
3) With high samplerates (like a 44100 samplerate BGM pitched with values >100) audio thread is not fast enough to correctly stream the song resulting in a corrupted BGM playback.

**Comment by Rinnegatamante** [comment]
BGM_Pitch finally fixed :P

**Comment by gnmmarechal** [comment]
awesome, updated the testbuild for the updater with the latest commit.

**Comment by Ghabry** [comment]
You wrote a while ago that OGG does not work correctly.
Maybe this helps: https://github.com/Ghabry/easyrpg-player/commit/9ef84698b6aacca8e563d1666521956740494251#diff-5ae40f1d5e418f1297a3df932be3d5b7L24

**Comment by Rinnegatamante** [comment]
Not at all, OGG doesn't work correctly only for BGMs cause it needs a different audio streaming function (since vorbisidec doesn't give you static PCM buffer sizes) and cause apparently secondary thread just crashes when used for BGM oggs. Sounds are working perfectly.
